src/components/projecteditor/compiler.js

```diff
@@ -229,7 +230,12 @@ export default class Compiler extends Component {
                                 });
                             }
                             else {
-                                this.consoleRows.push({channel:2,msg:"[ERROR] The contract " + contractName + " could not be compiled."});
+                                if (data.contracts) {
+                                    this.consoleRows.push({channel:2,msg:"[ERROR] The contract " + contractName + " could not be compiled. The contract needs to be named same as the contract source file."});
```

Comment:
the text should read `The contract " + contractName + " could not be compiled. The contract needs to be named the same as the contract's source file."